### PR TITLE
feat(dpp)!: do not allow creating data contracts without documents

### PR DESCRIPTION
**Pull request by markin-io**
## Issue being fixed or feature implemented

`check_tx` now will throw an error in case data contract schema is empty

## What was done?

- Added new error type
- Added wasm binings

## How Has This Been Tested?

- Added new test to test-suite

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [ ] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/platform-test-suite/test/functional/platform/DataContract.spec.js [hunk 1]
```diff
@@ -39,6 +40,23 @@ describe('Platform', () => {
       }
     });
 
+    it('should fail to create new data contract with empty schema', async () => {
```
Comment (shumkov) [inline packages/platform-test-suite/test/functional/platform/DataContract.spec.js:43]: Do you think we need a test for this in test suite? Maybe just in DPP?

File: packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/data_contract_create/structure/v0/mod.rs [hunk 2]
```diff
@@ -17,6 +20,12 @@ impl DataContractCreatedStateTransitionStructureValidationV0 for DataContractCre
...
         platform_version: &PlatformVersion,
     ) -> Result<SimpleConsensusValidationResult, Error> {
+        if self.data_contract().document_schemas().is_empty() {
+            return Ok(SimpleConsensusValidationResult::new_with_error(
+                DataContractEmptySchemaError::new(self.data_contract().id().clone()).into(),
+            ));
+        }
+
         // Validate data contract
         let result = DataContract::try_from_platform_versioned(
```
Comment (shumkov) [inline packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/data_contract_create/structure/v0/mod.rs:30]: Actual schema validation is happening here for each document type https://github.com/dashpay/platform/blob/10954c2355903c42ff17a2c3f7f57a5f674621ce/packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/mod.rs#L11

We need to validate here that we document types are defined https://github.com/dashpay/platform/blob/10954c2355903c42ff17a2c3f7f57a5f674621ce/packages/rs-dpp/src/data_contract/document_type/class_methods/create_document_types_from_document_schemas/mod.rs#L35

**Review by shumkov — CHANGES_REQUESTED** [review]

**Comment by thephez** [comment]
Technically this is a breaking change, right?

File: packages/platform-test-suite/test/functional/platform/DataContract.spec.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ const {
     StateTransitionBroadcastError,
   },
   PlatformProtocol: {
+    DataContractEmptySchemaError,
```
Comment (shumkov) [inline packages/platform-test-suite/test/functional/platform/DataContract.spec.js:13]: Never used

**Review by shumkov — CHANGES_REQUESTED** [review]

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**